### PR TITLE
Add bounds checking to avoid Out-of-bounds Write for gif2h5

### PR DESCRIPTION
**Pull request by wswsmao**
fix CVE-2022-25972
https://nvd.nist.gov/vuln/detail/CVE-2022-25972

### CODE REVIEW

**Review by hyoklee — APPROVED** [review]

File: hl/tools/gif2h5/decompress.c [hunk 1]
```diff
@@ -296,6 +296,10 @@ Decompress(GIFIMAGEDESC *GifImageDesc, GIFHEAD *GifHead)
              * Build the hash table on-the-fly. No table is stored in the
              * file.
              */
+            if (FreeCode >= 4096) {
```
Comment (mattjala) [inline hl/tools/gif2h5/decompress.c:299]: Prefix/Suffix buffer size is `4096 * sizeof(int)`, so this should probably be `4096 * sizeof(int)` too.

Comment (jhendersonHDF) [reply hl/tools/gif2h5/decompress.c:299]: In this case the check would be correct since `FreeCode` is an index value into the `int` arrays

**Review by mattjala — APPROVED** [review]

**Comment by derobins** [comment]
Do you have a test file? There's no test to ensure this is really fixed.

**Comment by wswsmao** [comment]
> Do you have a test file? There's no test to ensure this is really fixed.

Ok，I can add test files as appendix

**Pull request merged**